### PR TITLE
collect workflows and subworkflows in branch nodes

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: cosmicBboy <niels.bantilan@gmail.com>

# TL;DR

Fixes https://github.com/flyteorg/flyte/issues/781

This diff updates `SdkWorkflow.get_sub_workflows` so that it collects workflows/subworkflows from conditional branches.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

See https://github.com/flyteorg/flyte/issues/781 for more details.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/781

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]
+1 but also confirm with flytesnacks?  the broken example from joseph.

**Comment by cosmicBboy** [comment]
> +1 but also confirm with flytesnacks? the broken example from joseph.

@wild-endeavor haven't been able to successfully test this change since `make -C cookbook/core register` fails on me, I'll wait until I have aws access

**Comment by wild-endeavor** [comment]
sorry, was in meetings.  Try just `make register` - i gave you the wrong command, sorry.

**Review by wild-endeavor — APPROVED** [review]
+1 assuming joseph's example works.

**Comment by cosmicBboy** [comment]
cool, `make register` worked!

For the record:

- added `cookbook/core/basic/conditional.py` with the code:
```python
from flytekit import task, workflow
from flytekit.core.condition import conditional


@task
def t() -> int:
    return 5


@workflow
def wf1() -> int:
    return t()


@workflow
def branching(x: int) -> int:
    return (
        conditional("test").if_(x == 2).then(wf1()).else_().fail("unexpected")
    )

```

`make register` result:

```
...

Registered resource_type: WORKFLOW
project: "flytesnacks"
domain: "development"
name: "core.basic.conditional.branching"
version: "34987222d10d71144e7beba5b73e3e1f67ef1a4a"

Registered resource_type: LAUNCH_PLAN
project: "flytesnacks"
domain: "development"
name: "core.basic.conditional.branching"
version: "34987222d10d71144e7beba5b73e3e1f67ef1a4a"

...
```
